ciongke/Makefile

```diff
@@ -0,0 +1,138 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# These are the usual GKE variables. Project is purposefully empty for now.
+PROJECT = k8s-ci-on-gke
+ZONE = us-central1-b
+NUM_NODES = 6
+MACHINE_TYPE = n1-standard-8
+
+# These are GitHub credentials.
+# The hook secret is your HMAC token, the OAuth secret is the OAuth
+# token of whatever account you want to comment and update statuses.
+HOOK_SECRET_FILE = $(shell echo $${HOME}/hook)
+OAUTH_SECRET_FILE = $(shell echo $${HOME}/k8s-oauth-token)
+ORG = kubernetes
+# This is test-infra-maintainers.
+TEAM = 2009229
+# If DRY_RUN is true then we won't make any mutating GitHub calls.
+DRY_RUN = true
+
+HOOK_VERSION = 0.0
+
+# Useful rules:
+# - create-cluster turns up a cluster then prints out the webhook address.
+# - delete-cluster deletes the cluster.
+# - hook-image builds and pushes the latest hook image.
+# - hook-deployment updates the deployment.
+# - hook-service create the hook service.
+
+create-cluster:
+	gcloud -q container --project "$(PROJECT)" clusters create ciongke --zone "$(ZONE)" --machine-type "$(MACHINE_TYPE)" --scope "https://www.googleapis.com/auth/compute","https://www.googleapis.com/auth/devstorage.full_control","https://www.googleapis.com/auth/logging.write","https://www.googleapis.com/auth/servicecontrol","https://www.googleapis.com/auth/service.management" --num-nodes "$(NUM_NODES)" --network "default" --enable-cloud-logging --enable-cloud-monitoring
+	kubectl create secret generic hmac-token --from-file=hmac=$(HOOK_SECRET_FILE)
+	kubectl create secret generic oauth-token --from-file=oauth=$(OAUTH_SECRET_FILE)
+	make hook-image --no-print-directory
+	make hook-deployment --no-print-directory
+	make hook-service --no-print-directory
+	@echo -n "Waiting for loadbalancer ingress "; while [[ "$$(kubectl get svc hook -o=json | jq -r .status.loadBalancer.ingress[0].ip)" == "null" ]]; do echo -n "."; sleep 5; done; echo " Done"
+	@echo "Webhook address: http://$$(kubectl get svc hook -o=json | jq -r .status.loadBalancer.ingress[0].ip):8888/"
+
+delete-cluster:
+	@echo "Deleting cluster in $(PROJECT) in 5 seconds..."; sleep 5
+	gcloud -q container --project "$(PROJECT)" clusters delete ciongke --zone "$(ZONE)"
+
+clean:
+	rm cmd/hook/hook
+
+.PHONY: create-cluster delete-cluster clean
+
+hook-image:
+	CGO_ENABLED=0 go build -o cmd/hook/hook github.com/kubernetes/test-infra/ciongke/cmd/hook
+	docker build -t "gcr.io/$(PROJECT)/hook:$(HOOK_VERSION)" cmd/hook
+	gcloud docker push "gcr.io/$(PROJECT)/hook:$(HOOK_VERSION)"
+
+hook-deployment:
+	@echo "$${hook_deployment}" | kubectl apply -f -
+
+hook-service:
+	@echo "$${hook_service}" | kubectl create -f -
+
+.PHONY: hook-image hook-deployment hook-service
+
+# We need to replace project and version and so on in these YAML definitions,
+# and there's not really a great way to do that.
+define hook_deployment
```

Comment:
can you use sed templating idea? or split this out into a separate file? bash heredocs would work too